bigquery/tests/system.py

```diff
@@ -1467,19 +1467,24 @@ def test_nested_table_to_dataframe(self):
             'nested_record': {'nested_nested_string': 'some deep insight'},
         }
         to_insert = [
-            ('Some value', record)
+            {'string_col': 'Some value', 'record_col': record},
         ]
+        rows = [json.dumps(row) for row in to_insert]
+        body = six.StringIO('{}\n'.format('\n'.join(rows)))
         table_id = 'test_table'
         dataset = self.temp_dataset(_make_dataset_id('nested_df'))
-        table_arg = Table(dataset.table(table_id), schema=schema)
-        table = retry_403(Config.CLIENT.create_table)(table_arg)
+        table = dataset.table(table_id)
         self.to_delete.insert(0, table)
-        Config.CLIENT.create_rows(table, to_insert)
-        QUERY = 'SELECT * from `{}.{}.{}`'.format(
-            Config.CLIENT.project, dataset.dataset_id, table_id)
-
-        retry = RetryResult(_has_rows, max_tries=8)
-        df = retry(self._fetch_dataframe)(QUERY)
+        job_config = bigquery.LoadJobConfig()
+        job_config.write_disposition = 'WRITE_TRUNCATE'
+        job_config.source_format = 'NEWLINE_DELIMITED_JSON'
+        job_config.schema = schema
+        # Load a table using a local JSON file from memory.
+        Config.CLIENT.load_table_from_file(
+            body, table, job_config=job_config).result()
+
+        df = Config.CLIENT.list_rows(
+            table, selected_fields=schema).to_dataframe()
```

Comment:
What drives such a big change to this testcase?  Wouldn't replacing `create_rows` with `insert_rows` just work?

Reply:
In short, the reason is to reduce unnecessary test flakiness.

It would have continued to work as well as it had been, but `create_rows` / `insert_rows` is not intended for this use case (writing rows to a just-created empty table). After you create a table there is a time period (up to 2 minutes) where the streaming API might not be able to write to it due to some internal eventual consistency / cacheing. A load job does not have these same concerns.

Reply:
OK, thanks.  In a future PR, it might be good to note the "I found this other thing while renaming" bit to the description.